src/main/java/duke/utilities/Parser.java

```diff
@@ -0,0 +1,99 @@
+package duke.utilities;
+
+import duke.commands.*;
+import duke.tasks.*;
+import duke.exceptions.*;
+import java.util.regex.Pattern;
+
+public class Parser {
+    public Parser() {
+
+    }
+
+    public static String parseTask(Task task) { // parses task into string to load into storage
+        String parsed;
+        TaskType taskType = task.getTaskType();
+        switch (taskType) {
+            case TODO:
+                parsed = taskType.toString() + "|" + task.getDoneInt() + "|" + task.getDescription();
+                break;
+            case EVENT:
+                parsed = taskType.toString() + "|" + task.getDoneInt() + "|" + task.getDescription() + "|" + ((Event) task).getTaskTime();
+                break;
+            case DEADLINE:
+                parsed = taskType.toString() + "|" + task.getDoneInt() + "|" + task.getDescription() + "|" + ((Deadline) task).getTaskTime();
+                break;
+            default:
+                parsed = ""; // TODO add error message, change duke.exceptions.DukeException to such task exists
+        }
+        return parsed;
+    }
+
+    public static Task parseFile(String line) { // parses line from tasks.txt into a task
+        String[] split = line.split(Pattern.quote("|"));
+        switch (split[0]) {
+            case "T":
+                return new ToDo(split[1], split[2]);
+            case "E":
+                return new Event(split[1], split[2], split[3]);
+            case "D":
+                return new Deadline(split[1], split[2], split[3]);
+            default:
+                System.out.println("Error in parser parse method"); // TODO add error message and duke.exceptions.DukeException
+                return null;
+        }
+    }
+
+    public static Command parseInput(String input) throws DukeException {
+        try {
+            input = input.trim(); // trim any opening and trailing whitespace
+            if (input.equals("bye")) { // parse exit command
+                return new ExitCommand();
+            }
+            else if (input.equals("list")) { // parse list command
```

Comment:
Would the egyptian style be better for this part? Where the else if statements come directly after } brackets.